code/game/machinery/cryopod.dm

```diff
@@ -333,23 +334,37 @@ obj/machinery/computer/cryopod/Topic(href, href_list)
 				message_admins("\blue [key_name_admin(M)] has entered a stasis pod.")
 
 				//Despawning occurs when process() is called with an occupant without a client.
-				src.add_fingerprint(M)
+				add_fingerprint(M)
 
 /obj/machinery/cryopod/verb/eject()
 	set name = "Eject Pod"
 	set category = "Object"
 	set src in oview(1)
-	if(usr.stat != CONSCIOUS)
+	if(usr.stat != CONSCIOUS || !occupant)
+		return
+
+	//You can eject:
+	// - yourself
+	// - not clientless occupants
+	// - dead occupants
+	// - occupants in critical condition
+	// - anyone if you are head or officer
+	if(usr != occupant && \
```

Comment:
```
	if(usr != occupant \
		  && !occupant.client \
		  && occupant.stat != DEAD \
		  && occupant.health - occupant.halloss > config.health_threshold_softcrit \
		  && !allowed(usr))
		to_chat(usr, "<span class='red'>You can't eject person from [src], since the preservation procedure has already begun</span>")
		return
```
Ну и комментарии не нужны, тут всё достаточно прозрачно.